### ISSUE TITLE
[REQUEST] EmberZNet NCP 7.3.x.x Zigbee Coordinator NCP firmware image for Sonoff Zigbee 3.0 USB Dongle Plus V2 (model “ZBDongle-E”) by ITead?

### ISSUE THREAD

**Issue by Hedda** [itead/Sonoff_Zigbee_Dongle_Firmware#31]
Follow up to https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/issues/28 for tracking as separate issue because also need a EmberZNet NCP 7.3.x.x firmware image for Zigbee2MQTT migration.

@Daniel-zhan-itead @liangjia2019 @guozi7788 I like to recommend that ITead releasing an official Zigbee NCP firmware image based on EmberZNet NCP 7.3.x.0 version or later 7.3.x.x bug-fix version for your EFR32MG21 based  [“Sonoff Zigbee 3.0 USB Dongle Plus V2”](https://itead.cc/product/zigbee-3-0-usb-dongle/) (model [“ZBDongle-E”, a.k.a. "Dongle-E”](https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-e/)) as an optional firmware update and required for Zigbee2MQTT upgrade path (because existing Zigbee2MQTT users need to first upgrade to EmberZNet NCP 7.3 before thay can migrate to their new driver that support EmberZNet NCP 7.4).

Please also build and release Silabs Zigbee EmberZNet SDK 7.3.3.0 firmware image from Silicon Labs Gecko SDK (GSDK) Version 4.3.3.0

* https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-7.3.3.0.pdf
* https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.3.3

Without that EmberZNet NCP 7.3 version then Z2M users can not backup their Zigbee network for migration restore to 7.4 version:

* https://community.home-assistant.io/t/ezsp-to-ember/732705

That is, I forgot to mention it in https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/issues/28 but you also need to release EmberZNet Zigbee NCP 7.3.x.0 version as well for smooth migration, at least for Zigbee2MQTT migration when already running EmberZNet Zigbee NCP versions older than 7.3.x.x

Note! It should be EmberZNet Zigbee NCP 7.3.x.x (and EmberZNet Zigbee NCP 7.4.x.x) firmware with software flow control only.

Suggest check out darkxst's config -> https://github.com/darkxst/silabs-firmware-builder/blob/main/manifests/zbdonglee.json

As I understand, while you need to have upgraded to at least EmberZNet 7.4.x.x firmware version or later if you want to install Zigbee2MQTT from scratch today with their new ”ember” driver/adapter in Zigbee2MQTT, from what I read the recommendation is slighly different if already running Zigbee2MQTT with their old ”ezsp” driver/adapter in Zigbee2MQTT.

If want to upgrade when using Zigbee2MQTT then it is recommended to first upgrade to EmberZNet NCP 7.3.x.x firmware version while still using the old ”ezsp” driver/adapter as it sounds like migration to the new ”ember” driver will go smoother if already have had the old ”ezsp” driver/adapter working EmberZNet NCP 7.3.x firmware version while still using the old ”ezsp” driver/adapter in Zigbee2MQTT. Only after already upgraded to EmberZNet NCP 7.3.x.x firmware version and get it working with that should existing Zigbee2MQTT users upgrade to EmberZNet NCP 7.3.x firmware version.

* https://www.zigbee2mqtt.io/guide/adapters/emberznet.html

Suggest read and ask more specific questions in Zigbee2MQTT’s own community forum on GitHub after you done more research. The multi-step process you need to do depends on your specific setup and you need to ask in the Zigbee2MQTT community if you need step-by-step that applies to your setup. The process for actually updating firmware version will not be different.

Suggest read and try following this article regarding migrating Zigbee2MQTT from the `ezsp` adapter to the new `ember` adapter:

- https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/

If you require assistance with Zigbee2MQTT then report as issue or discuss directly on Zigbee2MQTT's repository on GitHub:

* https://github.com/Koenkk/zigbee2mqtt/issues
* https://github.com/Koenkk/zigbee2mqtt/discussions
* or Discord at https://discord.gg/NyseBeK

Summery; `ezsp` adapter is now deprecated in Zigbee2MQTT’s zigbee-herdsman library, which means that the Zigbee2MQTT developers are by doing so strongly recommending users to upgrade and migrate to `ember` adapter, and that migration might require a multi-step process depending on what EmberZNet firmware version you are using today. Users that continues to use the `ezsp` adapter will for now simply get a warning in their logs that they are using a deprecated adapter, however it will not stop working.

* https://www.zigbee2mqtt.io/guide/adapters/
  * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html

For reference see:

* https://github.com/Koenkk/zigbee-herdsman/pull/918
* https://github.com/Koenkk/zigbee2mqtt/discussions/21462
  * https://github.com/Koenkk/zigbee2mqtt/discussions/21462#discussioncomment-8510320
* https://github.com/Koenkk/zigbee-herdsman/pull/1039

PS: "Deprecated" does not mean that the `ezsp` adapter driver in Zigbee2MQTT will be removed yet, instead it means you are discouraged from using it in in Zigbee2MQTT configuration as it is being phased out and replaced by the newer `ember` adapter driver (which is not compatible with older firmware or all hardware).

* https://en.wikipedia.org/wiki/Deprecation#Software
* https://techterms.com/definition/deprecated
* https://www.baeldung.com/cs/deprecated-vs-obsolete[Zigbee EmberZNet SDK](https://www.silabs.com/wireless/zigbee#software)	7.3.3.0

**Comment by liangjia2019** [comment]
> Follow up to #28 for tracking as separate issue because also need a EmberZNet NCP 7.3.x.x firmware image for Zigbee2MQTT migration.
> 
> @Daniel-zhan-itead @liangjia2019 @guozi7788 I like to recommend that ITead releasing an official Zigbee NCP firmware image based on EmberZNet NCP 7.3.x.0 version or later 7.3.x.x bug-fix version for your EFR32MG21 based [“Sonoff Zigbee 3.0 USB Dongle Plus V2”](https://itead.cc/product/zigbee-3-0-usb-dongle/) (model [“ZBDongle-E”, a.k.a. "Dongle-E”](https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-e/)) as an optional firmware update and required for Zigbee2MQTT upgrade path (because existing Zigbee2MQTT users need to first upgrade to EmberZNet NCP 7.3 before thay can migrate to their new driver that support EmberZNet NCP 7.4).
> 
> Please also build and release Silabs Zigbee EmberZNet SDK 7.3.3.0 firmware image from Silicon Labs Gecko SDK (GSDK) Version 4.3.3.0
> 
>     * https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-7.3.3.0.pdf
> 
>     * https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.3.3
> 
> 
> Without that EmberZNet NCP 7.3 version then Z2M users can not backup their Zigbee network for migration restore to 7.4 version:
> 
>     * https://community.home-assistant.io/t/ezsp-to-ember/732705
> 
> 
> That is, I forgot to mention it in #28 but you also need to release EmberZNet Zigbee NCP 7.3.x.0 version as well for smooth migration, at least for Zigbee2MQTT migration when already running EmberZNet Zigbee NCP versions older than 7.3.x.x
> 
> Note! It should be EmberZNet Zigbee NCP 7.3.x.x (and EmberZNet Zigbee NCP 7.4.x.x) firmware with software flow control only.
> 
> Suggest check out darkxst's config -> https://github.com/darkxst/silabs-firmware-builder/blob/main/manifests/zbdonglee.json
> 
> As I understand, while you need to have upgraded to at least EmberZNet 7.4.x.x firmware version or later if you want to install Zigbee2MQTT from scratch today with their new ”ember” driver/adapter in Zigbee2MQTT, from what I read the recommendation is slighly different if already running Zigbee2MQTT with their old ”ezsp” driver/adapter in Zigbee2MQTT.
> 
> If want to upgrade when using Zigbee2MQTT then it is recommended to first upgrade to EmberZNet NCP 7.3.x.x firmware version while still using the old ”ezsp” driver/adapter as it sounds like migration to the new ”ember” driver will go smoother if already have had the old ”ezsp” driver/adapter working EmberZNet NCP 7.3.x firmware version while still using the old ”ezsp” driver/adapter in Zigbee2MQTT. Only after already upgraded to EmberZNet NCP 7.3.x.x firmware version and get it working with that should existing Zigbee2MQTT users upgrade to EmberZNet NCP 7.3.x firmware version.
> 
>     * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html
> 
> 
> Suggest read and ask more specific questions in Zigbee2MQTT’s own community forum on GitHub after you done more research. The multi-step process you need to do depends on your specific setup and you need to ask in the Zigbee2MQTT community if you need step-by-step that applies to your setup. The process for actually updating firmware version will not be different.
> 
> Suggest read and try following this article regarding migrating Zigbee2MQTT from the `ezsp` adapter to the new `ember` adapter:
> 
>     * https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/
> 
> 
> If you require assistance with Zigbee2MQTT then report as issue or discuss directly on Zigbee2MQTT's repository on GitHub:
> 
>     * https://github.com/Koenkk/zigbee2mqtt/issues
> 
>     * https://github.com/Koenkk/zigbee2mqtt/discussions
> 
>     * or Discord at https://discord.gg/NyseBeK
> 
> 
> Summery; `ezsp` adapter is now deprecated in Zigbee2MQTT’s zigbee-herdsman library, which means that the Zigbee2MQTT developers are by doing so strongly recommending users to upgrade and migrate to `ember` adapter, and that migration might require a multi-step process depending on what EmberZNet firmware version you are using today. Users that continues to use the `ezsp` adapter will for now simply get a warning in their logs that they are using a deprecated adapter, however it will not stop working.
> 
>     * https://www.zigbee2mqtt.io/guide/adapters/
>       
>       * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html
> 
> 
> For reference see:
> 
>     * [New ember adapter implementation targeting EZSP 13 and above Koenkk/zigbee-herdsman#918](https://github.com/Koenkk/zigbee-herdsman/pull/918)
> 
>     * [Improved Ember (EZSP) driver (experimental) Koenkk/zigbee2mqtt#21462](https://github.com/Koenkk/zigbee2mqtt/discussions/21462)
>       
>       * [Improved Ember (EZSP) driver (experimental) Koenkk/zigbee2mqtt#21462 (comment)](https://github.com/Koenkk/zigbee2mqtt/discussions/21462#discussioncomment-8510320)
> 
>     * [feat: Log EZSP adapter deprecated Koenkk/zigbee-herdsman#1039](https://github.com/Koenkk/zigbee-herdsman/pull/1039)
> 
> 
> PS: "Deprecated" does not mean that the `ezsp` adapter driver in Zigbee2MQTT will be removed yet, instead it means you are discouraged from using it in in Zigbee2MQTT configuration as it is being phased out and replaced by the newer `ember` adapter driver (which is not compatible with older firmware or all hardware).
> 
>     * https://en.wikipedia.org/wiki/Deprecation#Software
> 
>     * https://techterms.com/definition/deprecated
> 
>     * [https://www.baeldung.com/cs/deprecated-vs-obsolete[Zigbee](https://www.baeldung.com/cs/deprecated-vs-obsolete%5BZigbee) EmberZNet SDK](https://www.silabs.com/wireless/zigbee#software)	7.3.3.0

Hi 

> Follow up to #28 for tracking as separate issue because also need a EmberZNet NCP 7.3.x.x firmware image for Zigbee2MQTT migration.
> 
> @Daniel-zhan-itead @liangjia2019 @guozi7788 I like to recommend that ITead releasing an official Zigbee NCP firmware image based on EmberZNet NCP 7.3.x.0 version or later 7.3.x.x bug-fix version for your EFR32MG21 based [“Sonoff Zigbee 3.0 USB Dongle Plus V2”](https://itead.cc/product/zigbee-3-0-usb-dongle/) (model [“ZBDongle-E”, a.k.a. "Dongle-E”](https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-e/)) as an optional firmware update and required for Zigbee2MQTT upgrade path (because existing Zigbee2MQTT users need to first upgrade to EmberZNet NCP 7.3 before thay can migrate to their new driver that support EmberZNet NCP 7.4).
> 
> Please also build and release Silabs Zigbee EmberZNet SDK 7.3.3.0 firmware image from Silicon Labs Gecko SDK (GSDK) Version 4.3.3.0
> 
>     * https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-7.3.3.0.pdf
> 
>     * https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.3.3
> 
> 
> Without that EmberZNet NCP 7.3 version then Z2M users can not backup their Zigbee network for migration restore to 7.4 version:
> 
>     * https://community.home-assistant.io/t/ezsp-to-ember/732705
> 
> 
> That is, I forgot to mention it in #28 but you also need to release EmberZNet Zigbee NCP 7.3.x.0 version as well for smooth migration, at least for Zigbee2MQTT migration when already running EmberZNet Zigbee NCP versions older than 7.3.x.x
> 
> Note! It should be EmberZNet Zigbee NCP 7.3.x.x (and EmberZNet Zigbee NCP 7.4.x.x) firmware with software flow control only.
> 
> Suggest check out darkxst's config -> https://github.com/darkxst/silabs-firmware-builder/blob/main/manifests/zbdonglee.json
> 
> As I understand, while you need to have upgraded to at least EmberZNet 7.4.x.x firmware version or later if you want to install Zigbee2MQTT from scratch today with their new ”ember” driver/adapter in Zigbee2MQTT, from what I read the recommendation is slighly different if already running Zigbee2MQTT with their old ”ezsp” driver/adapter in Zigbee2MQTT.
> 
> If want to upgrade when using Zigbee2MQTT then it is recommended to first upgrade to EmberZNet NCP 7.3.x.x firmware version while still using the old ”ezsp” driver/adapter as it sounds like migration to the new ”ember” driver will go smoother if already have had the old ”ezsp” driver/adapter working EmberZNet NCP 7.3.x firmware version while still using the old ”ezsp” driver/adapter in Zigbee2MQTT. Only after already upgraded to EmberZNet NCP 7.3.x.x firmware version and get it working with that should existing Zigbee2MQTT users upgrade to EmberZNet NCP 7.3.x firmware version.
> 
>     * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html
> 
> 
> Suggest read and ask more specific questions in Zigbee2MQTT’s own community forum on GitHub after you done more research. The multi-step process you need to do depends on your specific setup and you need to ask in the Zigbee2MQTT community if you need step-by-step that applies to your setup. The process for actually updating firmware version will not be different.
> 
> Suggest read and try following this article regarding migrating Zigbee2MQTT from the `ezsp` adapter to the new `ember` adapter:
> 
>     * https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/
> 
> 
> If you require assistance with Zigbee2MQTT then report as issue or discuss directly on Zigbee2MQTT's repository on GitHub:
> 
>     * https://github.com/Koenkk/zigbee2mqtt/issues
> 
>     * https://github.com/Koenkk/zigbee2mqtt/discussions
> 
>     * or Discord at https://discord.gg/NyseBeK
> 
> 
> Summery; `ezsp` adapter is now deprecated in Zigbee2MQTT’s zigbee-herdsman library, which means that the Zigbee2MQTT developers are by doing so strongly recommending users to upgrade and migrate to `ember` adapter, and that migration might require a multi-step process depending on what EmberZNet firmware version you are using today. Users that continues to use the `ezsp` adapter will for now simply get a warning in their logs that they are using a deprecated adapter, however it will not stop working.
> 
>     * https://www.zigbee2mqtt.io/guide/adapters/
>       
>       * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html
> 
> 
> For reference see:
> 
>     * [New ember adapter implementation targeting EZSP 13 and above Koenkk/zigbee-herdsman#918](https://github.com/Koenkk/zigbee-herdsman/pull/918)
> 
>     * [Improved Ember (EZSP) driver (experimental) Koenkk/zigbee2mqtt#21462](https://github.com/Koenkk/zigbee2mqtt/discussions/21462)
>       
>       * [Improved Ember (EZSP) driver (experimental) Koenkk/zigbee2mqtt#21462 (comment)](https://github.com/Koenkk/zigbee2mqtt/discussions/21462#discussioncomment-8510320)
> 
>     * [feat: Log EZSP adapter deprecated Koenkk/zigbee-herdsman#1039](https://github.com/Koenkk/zigbee-herdsman/pull/1039)
> 
> 
> PS: "Deprecated" does not mean that the `ezsp` adapter driver in Zigbee2MQTT will be removed yet, instead it means you are discouraged from using it in in Zigbee2MQTT configuration as it is being phased out and replaced by the newer `ember` adapter driver (which is not compatible with older firmware or all hardware).
> 
>     * https://en.wikipedia.org/wiki/Deprecation#Software
> 
>     * https://techterms.com/definition/deprecated
> 
>     * [https://www.baeldung.com/cs/deprecated-vs-obsolete[Zigbee](https://www.baeldung.com/cs/deprecated-vs-obsolete%5BZigbee) EmberZNet SDK](https://www.silabs.com/wireless/zigbee#software)	7.3.3.0

Hi，I have updated the NCP firmware for 7.3.3 and 7.4.3

**Comment by liangjia2019** [comment]
> Follow up to #28 for tracking as separate issue because also need a EmberZNet NCP 7.3.x.x firmware image for Zigbee2MQTT migration.
> 
> @Daniel-zhan-itead @liangjia2019 @guozi7788 I like to recommend that ITead releasing an official Zigbee NCP firmware image based on EmberZNet NCP 7.3.x.0 version or later 7.3.x.x bug-fix version for your EFR32MG21 based [“Sonoff Zigbee 3.0 USB Dongle Plus V2”](https://itead.cc/product/zigbee-3-0-usb-dongle/) (model [“ZBDongle-E”, a.k.a. "Dongle-E”](https://sonoff.tech/product/gateway-and-sensors/sonoff-zigbee-3-0-usb-dongle-plus-e/)) as an optional firmware update and required for Zigbee2MQTT upgrade path (because existing Zigbee2MQTT users need to first upgrade to EmberZNet NCP 7.3 before thay can migrate to their new driver that support EmberZNet NCP 7.4).
> 
> Please also build and release Silabs Zigbee EmberZNet SDK 7.3.3.0 firmware image from Silicon Labs Gecko SDK (GSDK) Version 4.3.3.0
> 
>     * https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-7.3.3.0.pdf
> 
>     * https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.3.3
> 
> 
> Without that EmberZNet NCP 7.3 version then Z2M users can not backup their Zigbee network for migration restore to 7.4 version:
> 
>     * https://community.home-assistant.io/t/ezsp-to-ember/732705
> 
> 
> That is, I forgot to mention it in #28 but you also need to release EmberZNet Zigbee NCP 7.3.x.0 version as well for smooth migration, at least for Zigbee2MQTT migration when already running EmberZNet Zigbee NCP versions older than 7.3.x.x
> 
> Note! It should be EmberZNet Zigbee NCP 7.3.x.x (and EmberZNet Zigbee NCP 7.4.x.x) firmware with software flow control only.
> 
> Suggest check out darkxst's config -> https://github.com/darkxst/silabs-firmware-builder/blob/main/manifests/zbdonglee.json
> 
> As I understand, while you need to have upgraded to at least EmberZNet 7.4.x.x firmware version or later if you want to install Zigbee2MQTT from scratch today with their new ”ember” driver/adapter in Zigbee2MQTT, from what I read the recommendation is slighly different if already running Zigbee2MQTT with their old ”ezsp” driver/adapter in Zigbee2MQTT.
> 
> If want to upgrade when using Zigbee2MQTT then it is recommended to first upgrade to EmberZNet NCP 7.3.x.x firmware version while still using the old ”ezsp” driver/adapter as it sounds like migration to the new ”ember” driver will go smoother if already have had the old ”ezsp” driver/adapter working EmberZNet NCP 7.3.x firmware version while still using the old ”ezsp” driver/adapter in Zigbee2MQTT. Only after already upgraded to EmberZNet NCP 7.3.x.x firmware version and get it working with that should existing Zigbee2MQTT users upgrade to EmberZNet NCP 7.3.x firmware version.
> 
>     * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html
> 
> 
> Suggest read and ask more specific questions in Zigbee2MQTT’s own community forum on GitHub after you done more research. The multi-step process you need to do depends on your specific setup and you need to ask in the Zigbee2MQTT community if you need step-by-step that applies to your setup. The process for actually updating firmware version will not be different.
> 
> Suggest read and try following this article regarding migrating Zigbee2MQTT from the `ezsp` adapter to the new `ember` adapter:
> 
>     * https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/
> 
> 
> If you require assistance with Zigbee2MQTT then report as issue or discuss directly on Zigbee2MQTT's repository on GitHub:
> 
>     * https://github.com/Koenkk/zigbee2mqtt/issues
> 
>     * https://github.com/Koenkk/zigbee2mqtt/discussions
> 
>     * or Discord at https://discord.gg/NyseBeK
> 
> 
> Summery; `ezsp` adapter is now deprecated in Zigbee2MQTT’s zigbee-herdsman library, which means that the Zigbee2MQTT developers are by doing so strongly recommending users to upgrade and migrate to `ember` adapter, and that migration might require a multi-step process depending on what EmberZNet firmware version you are using today. Users that continues to use the `ezsp` adapter will for now simply get a warning in their logs that they are using a deprecated adapter, however it will not stop working.
> 
>     * https://www.zigbee2mqtt.io/guide/adapters/
>       
>       * https://www.zigbee2mqtt.io/guide/adapters/emberznet.html
> 
> 
> For reference see:
> 
>     * [New ember adapter implementation targeting EZSP 13 and above Koenkk/zigbee-herdsman#918](https://github.com/Koenkk/zigbee-herdsman/pull/918)
> 
>     * [Improved Ember (EZSP) driver (experimental) Koenkk/zigbee2mqtt#21462](https://github.com/Koenkk/zigbee2mqtt/discussions/21462)
>       
>       * [Improved Ember (EZSP) driver (experimental) Koenkk/zigbee2mqtt#21462 (comment)](https://github.com/Koenkk/zigbee2mqtt/discussions/21462#discussioncomment-8510320)
> 
>     * [feat: Log EZSP adapter deprecated Koenkk/zigbee-herdsman#1039](https://github.com/Koenkk/zigbee-herdsman/pull/1039)
> 
> 
> PS: "Deprecated" does not mean that the `ezsp` adapter driver in Zigbee2MQTT will be removed yet, instead it means you are discouraged from using it in in Zigbee2MQTT configuration as it is being phased out and replaced by the newer `ember` adapter driver (which is not compatible with older firmware or all hardware).
> 
>     * https://en.wikipedia.org/wiki/Deprecation#Software
> 
>     * https://techterms.com/definition/deprecated
> 
>     * [https://www.baeldung.com/cs/deprecated-vs-obsolete[Zigbee](https://www.baeldung.com/cs/deprecated-vs-obsolete%5BZigbee) EmberZNet SDK](https://www.silabs.com/wireless/zigbee#software)	7.3.3.0

Hi Hedda:
     I have updated the NCP firmware for 7.3.3 and 7.4.3.

**Comment by cvocvo** [comment]
@liangjia2019 Are there any instructions for how to upgrade the firmware from whatever version shipped on my device to v7.4.3? I'm using ZHA in Home Assistant.
The link here: https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP_7.4.3#firmware-upgrade-and-recovery-procedure
Sends me here: https://sonoff.tech/product-document/gateway-and-sensors-doc/zigbee-dongle-plus-efr32mg21-doc/
Which doesn't seem to load the firmware document anymore. I just see the "Specification" and "User Manual" links.

**Comment by liangjia2019** [comment]
> @liangjia2019 Are there any instructions for how to upgrade the firmware from whatever version shipped on my device to v7.4.3? I'm using ZHA in Home Assistant. The link here: https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP_7.4.3#firmware-upgrade-and-recovery-procedure Sends me here: https://sonoff.tech/product-document/gateway-and-sensors-doc/zigbee-dongle-plus-efr32mg21-doc/ Which doesn't seem to load the firmware document anymore. I just see the "Specification" and "User Manual" links.

I think you can refer to the steps of zigbee2mqtt for introduction
The link here:https://www.zigbee2mqtt.io/devices/ZBDongle-E.html

**Comment by liangjia2019** [comment]
I think you can refer to the steps of zigbee2mqtt for introduction.
The link here:https://www.zigbee2mqtt.io/devices/ZBDongle-E.html

**Comment by liangjia2019** [comment]
> @liangjia2019 Are there any instructions for how to upgrade the firmware from whatever version shipped on my device to v7.4.3? I'm using ZHA in Home Assistant. The link here: https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP_7.4.3#firmware-upgrade-and-recovery-procedure Sends me here: https://sonoff.tech/product-document/gateway-and-sensors-doc/zigbee-dongle-plus-efr32mg21-doc/ Which doesn't seem to load the firmware document anymore. I just see the "Specification" and "User Manual" links.

In HA,you also add addon Silicon Labs Flasher
![图片](https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/assets/52771300/3aed539b-45fd-4440-a690-beac2f20e536)

**Comment by Hedda** [comment]
@cvocvo How-to flash/program a firmware image is off-topic for this specific issue and should really be dealt with in a separate issue in order not to thread-jack this specific request about an exact version, however, I am going to give links with context below to what I know, but I hope that you will then not continue that discussion elsewhere like start a new separate issue in this repository or alternative post in a existing forum discussion thread, like example:

* https://community.home-assistant.io/t/itead-s-sonoff-zigbee-3-0-usb-dongle-plus-v2-model-zbdongle-e-based-on-silicon-labs-efr32mg21-20dbm-radio-soc-mcu/442695


> Are there any instructions for how to upgrade the firmware from whatever version shipped on my device to v7.4.3? I'm using ZHA in Home Assistant.
> The link here: https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/tree/master/Dongle-E/NCP_7.4.3#firmware-upgrade-and-recovery-procedure
> Sends me here: https://sonoff.tech/product-document/gateway-and-sensors-doc/zigbee-dongle-plus-efr32mg21-doc/
> Which doesn't seem to load the firmware document anymore. I just see the "Specification" and "User Manual" links.

@Daniel-zhan-itead @liangjia2019 @guozi7788 best would be if you either see to that ITead/Sonoff add and maintain official user-manual/guide documentation to ITead/Sonoff webpages for how-to to flash firmware to ZBDongle-E or that you at least add and maintain links to trusted third-party tools and manuals/guides documentation:

* https://sonoff.tech/wp-content/uploads/2022/11/SONOFF-Zigbee-3.0-USB-dongle-plus-firmware-flashing-.pdf
* http://sonoff.tech/wp-content/uploads/2022/08/CE_Declaration_of_Conformity_ZBDongle-E.pdf
* https://sonoff.tech/product-document/gateway-and-sensors-doc/zigbee-dongle-plus-efr32mg21-doc/
* https://itead.cc/product/zigbee-3-0-usb-dongle/

Please also help update community webpages that accept documentation contribution from the community, like example these:

* https://zigbee.blakadder.com/Sonoff_ZBDongle-E.html
* https://www.zigbee2mqtt.io/devices/ZBDongle-E.html
* https://www.home-assistant.io/integrations/zha

> In HA,you can also add addon Silicon Labs Flasher

Note that I do not believe that Nabu Casa's upstream repositories of "Silicon Labs Flasher Add-on" and "sl-web-tools" tool (both of which are based on Nabu Casa 's "universal-silabs-flasher") contain code that has been tested for flashing/programming compatibility with any other adapter than Nabu Casa's own Home Assistant SkyConnect / Home Assistant Connect ZBT-T dongle, so probably not a good idea to refer owners of Sonoff ZBDongle-E to use Nabu Casa's upstream version of those as-is because the idea with those are they are open-source so other companies can either contribute to upstream or fork them if want to make sure it works with other adapters, as Nabu Casa currently only ensure that they are compatible with their own hardware as-is:

* https://github.com/home-assistant/addons/tree/master/silabs_flasher
* https://connectzbt1.home-assistant.io/firmware-update/#from-skyconnect
* https://www.home-assistant.io/blog/2023/02/08/state-of-matter-and-thread/#silabs-multi-flasher--sl-web-tools
* https://github.com/NabuCasa/sl-web-tools
* https://github.com/NabuCasa/universal-silabs-flasher

Therefor I believe it would at this time instead be better to just refer to either refer to third-party forks of those addons and flasher tools that have at least been tested by the community or to official tools by Silicon Labs (Silabs) that all allow you to pick and flash firmware from file.

See these examples with forks of above that added and tested support for other hardware, including some Sonoff  adapters:

* https://darkxst.github.io/silabs-firmware-builder/
* https://github.com/darkxst/sl-web-tools
* https://github.com/darkxst/multipan_flasher
* https://github.com/tube0013/tubeszb_addons/tree/main/tzb-silabs-flasher
* https://github.com/tube0013/tubeszb_addons/tree/main/tzb-cc2652-flasher
* https://github.com/mercenaruss/zigstar_addons/tree/main/zigstar-silabs-flasher
* https://github.com/mercenaruss/zigstar_addons/tree/main/zigstar-cc2652-flasher
* https://github.com/mercenaruss/zigstar_addons/tree/main/zigstar-cc2652p7-flasher
* https://smlight.tech/flasher/

And see these examples official documentation and tools from Silicon Labs about programming (flashing) and/or recovery (most of which unfortunatly require user to have a EFR32 compatible J-LInk debug probe programmer adapter hardware):

* https://www.silabs.com/developers/simplicity-studio/simplicity-commander
* https://www.silabs.com/developers/simplicity-studio
* https://github.com/SiliconLabs/simplicity_sdk
* https://www.silabs.com/documents/public/user-guides/ug162-simplicity-commander-reference-guide.pdf
* https://www.silabs.com/documents/public/application-notes/an1218-secure-boot-with-rtsl.pdf
* https://docs.silabs.com/simplicity-studio-5-users-guide/latest/ss-5-users-guide-overview/
* https://docs.silabs.com/simplicity-studio-5-users-guide/5.8.2/ss-5-users-guide-tools-overview/
* https://docs.silabs.com/simplicity-studio-5-users-guide/5.8.2/ss-5-users-guide-building-and-flashing/#flash-programmer
* https://docs.silabs.com/simplicity-studio-5-users-guide/5.8.2/ss-5-users-guide-building-and-flashing/flashing

PS: If you manually put the adapter into bootloader mode then you can flash ut with any software supporting the XMODEM file transfer protocol for sending the firmware image to the device. 

* https://en.wikipedia.org/wiki/XMODEM

**Comment by cvocvo** [comment]
@liangjia2019 -- Thanks, the darkxst Silabs web flasher + 7.4.3 firmware seems to have worked great. per @Hedda I'll open up a separate issue if I run into any issues with it. Have a nice day!
